File: .NET/Examples/WindowsDesktop/Components/ArchiveControl.cs
using System;
using System.IO;
using System.Windows.Forms;
using System.Xml.Linq;
using UnderAutomation.UniversalRobots;
using UnderAutomation.UniversalRobots.Files;

public partial class ArchiveControl : UserControl, IUserControl
{
    private UR _ur;

    public ArchiveControl(UR ur)
    {
        _ur = ur;

        InitializeComponent();
    }

    #region IUserControl
    public string Title => "Decompile programs and installation";

    public bool FeatureEnabled => true;

    public void PeriodicUpdate()
    {
    }

    public void OnClose() { }
    public void OnOpen()
    {
    }
    #endregion



    private void btnOpen_Click(object sender, EventArgs e)
    {
        dlgOpen.InitialDirectory = Path.Combine(Path.GetDirectoryName(typeof(MainForm).Assembly.Location), "Samples");
        if (dlgOpen.ShowDialog() == DialogResult.OK) Decompile(dlgOpen.FileName);
    }

    public void Decompile(string fullName)
    {
        XElement xml = null;

        if (fullName.StartsWith("/"))
        {
            // download file by SFTP from robot
            using (var stream = new MemoryStream())
            {
                _ur.Sftp.DownloadFile(fullName, stream);
                stream.Position = 0;
                xml = URArchive.Load(stream);
            }
        }
        else
        {
            // load local file
            xml = URArchive.Load(fullName);
        }

        lblFile.Text = fullName;
        browser.DocumentText = "<?xml version=\"1.0\"?>\r\n" + xml.ToString();
    }
}

File: UnderAutomation.UniversalRobots.Showcase.Forms.vb/Components/ArchiveControl.vb
Imports System
Imports System.IO
Imports System.Windows.Forms
Imports System.Xml.Linq
Imports UnderAutomation.UniversalRobots
Imports UnderAutomation.UniversalRobots.Files

Public Partial Class ArchiveControl
    Inherits UserControl
    Implements IUserControl
    Private _ur As UR

    Public Sub New(ur As UR)
        _ur = ur

        InitializeComponent()
    End Sub

#Region "IUserControl"
    Public ReadOnly Property Title As String Implements IUserControl.Title
        Get
            Return "Decompile programs and installation"
        End Get
    End Property

    Public ReadOnly Property FeatureEnabled As Boolean Implements IUserControl.FeatureEnabled
        Get
            Return True
        End Get
    End Property

    Public Sub PeriodicUpdate() Implements IUserControl.PeriodicUpdate
    End Sub

    Public Sub OnClose() Implements IUserControl.OnClose
    End Sub
    Public Sub OnOpen() Implements IUserControl.OnOpen
    End Sub
#End Region



    Private Sub btnOpen_Click(sender As Object, e As EventArgs)
        dlgOpen.InitialDirectory = Path.Combine(Path.GetDirectoryName(GetType(MainForm).Assembly.Location), "Samples")
        If dlgOpen.ShowDialog() = DialogResult.OK Then Decompile(dlgOpen.FileName)
    End Sub

    Public Sub Decompile(fullName As String)
        Dim xml As XElement = Nothing

        If fullName.StartsWith("/") Then
            ' download file by SFTP from robot
            Using stream = New MemoryStream()
                _ur.Sftp.DownloadFile(fullName, stream)
                stream.Position = 0
                xml = URArchive.Load(stream)
            End Using
        Else
            ' load local file
            xml = URArchive.Load(fullName)
        End If

        lblFile.Text = fullName
        browser.DocumentText = "<?xml version=""1.0""?>" & Microsoft.VisualBasic.Constants.vbCrLf & xml.ToString()
    End Sub
End Class
